src/function.rs

```diff
@@ -205,10 +206,42 @@ impl<'lua> Function<'lua> {
             Ok(Function(lua.pop_ref()))
         }
     }
+
+    /// Returns source and line number information about the function.
+    ///
+    /// Corresponds to the `>S` what mask for `lua_getinfo` when applied
+    /// to the function.
+    pub fn source(&self) -> FunctionSource {
+        let lua = self.0.lua;
+        unsafe {
+            let mut ar: ffi::lua_Debug = std::mem::zeroed();
+            lua.push_ref(&self.0);
+            mlua_assert!(
+                ffi::lua_getinfo(lua.state, cstr!(">S"), &mut ar as *mut ffi::lua_Debug) != 0,
+                "lua_getinfo failed with `>S`"
+            );
+            FunctionSource {
+                source: String::from_utf8_lossy(CStr::from_ptr(ar.source as *const i8).to_bytes()).into_owned(),
+                short_src: String::from_utf8_lossy(CStr::from_ptr(&ar.short_src as *const i8).to_bytes()).into_owned(),
+                line_defined: ar.linedefined as i32,
+                last_line_defined: ar.lastlinedefined as i32,
+                what: String::from_utf8_lossy(CStr::from_ptr(ar.what as *const i8).to_bytes()).into_owned(),
+            }
+        }
+    }
 }
 
 impl<'lua> PartialEq for Function<'lua> {
     fn eq(&self, other: &Self) -> bool {
         self.0 == other.0
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct FunctionSource {
```

Comment:
Wondering, could the `FunctionSource` be confusing, since it represents mostly function information, rather than source text...
Maybe rename to `FunctionInfo` and the corresponding method to `info()` ?